Learn HTML setup and initial modules. (#8681)
GoogleChrome/web.dev: src/site/content/en/learn/html/headings-and-sections/index.md

In this section, you'll discover document structure; you'll recap the sectioning elements from the previous section; and mark up the outline for your application.

Choosing the right elements for the job as you code means you won't have to refactor or comment out your HTML. If you think about using the right element for the job, you'll most often pick the right element for the job. If you don't, you probably won't.

Now that you [understand markup semantics](/learn/html/semantic-html/)  and are aware of why choosing the right element is important, once you learn about all the different elements, you will generally pick the right element without much, if any, additional effort.    

## Site header

Let's build a site header. You'll start with non-semantic markup, and work your way to a good solution so that you can learn the benefits of the HTML section and heading elements along the way. 

If you put little to no thought into the semantics for our header, you might use code like this:

```html
<!-- start header --> 
<div id="pageHeader">
    <div id="title">Machine Learning Workshop</div>
    <!-- navigation -->
    <div id="navigation">
       <a href="#reg">Register</a>

>     <a href="#about">About</a>

Reviewer: ```suggestion
      <a href="#about">About</a>
```

       <a href="#teachers">Instructors</a>
       <a href="#feedback">Testimonials</a>
    </div>
    <!-- end navigation bar -->
</div> 

> <!-- end of header →

Reviewer: ```suggestion
<!-- end of header -->
```

  <address>Instructors: <a href="/hal.html">Hal</a> and <a href="/eve.html">Eve</a></address>
</footer>
```

### Document structure

This module starts with the `<header>` and `<footer>`, because they are unique in only sometimes being landmark, or "sectioning", elements. Let's cover the "full time" sectioning element by discussing the most common page layouts:

{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/LJs8vlTVDYBmM6t1z6rD.png", alt="A layout with a header, three columns, and a footer.", width="800", height="607" %}

A layout with a header, two sidebars, and a  footer, is known as the [holy grail layout](/patterns/layout/holy-grail/). There are many ways to mark up this content, including: 

```html  
<body>  
    <header>Header</header>  
    <nav>Nav</nav>  
    <main>Content</main>  
    <aside>Aside</aside>  
    <footer>Footer</footer>  
</body>  
```

If you are creating a blog, you might have a series of articles in `<main>`:

```html  
<body>  
    <header>Header</header>  
    <nav>Nav</nav>  
    <main>   
	<article>First post</article>  
	<article>Second post</article>  
    </main>  
    <aside>Aside</aside>  
    <footer>Footer</footer>  
</body>  
```

> Here, a `banner` and `contentinfo` are provided through a site `<header>` and `<footer>`.  The new elements added here include `<main>`, `<aside>`, and `<article>`; also, `<h1>` and `<nav>` that you used earlier, and `<section>`, which you haven't used yet.

Reviewer: ```suggestion
When employing semantic elements, browsers are able to create meaningful accessibility trees, enabling screen reader users to more easily navigate. Here, a `banner` and `contentinfo` are provided through a site `<header>` and `<footer>`.  The new elements added here include `<main>`, `<aside>`, and `<article>`; also, `<h1>` and `<nav>` that you used earlier, and `<section>`, which you haven't used yet.
```

Adding the first sentence explains why the image was included. It shows the banner and contentinfo.

An article isn't a landmark unless it has an accessible name; if it has an accessible name, the implicit role is `region`. [Landmark roles](https://developer.mozilla.org/docs/Web/Accessibility/ARIA/Roles#landmark_roles) should be used sparingly, to identify larger overall sections of the document. While using too many landmark roles can create "noise" in screen readers, making it difficult to understand the overall layout of the page, if your `<main>` contains two 10,000 word blog posts `<article>`s, including an accessible name for each article could be beneficial.

### `<section>`

> The `<section>` element is used to encompass generic standalone sections of a document when there is no more specific semantic element to use. Sections should have a heading, with very few exceptions.  Going back to the Jacinda Ardern example, on the home page of the newspaper, the banner would include the name of the newspaper, followed by a single `<main>`, divided into several `<section>`s, each with a header, such as "World news" and "Politics"; and in each section you'll find a series of `<article>`s. Within each `<article>`, you might find one or more `<section>` elements as well. If you look at this page, the entire "headers and sections'' part is the `<article>`. This `<article>` is then divided into several `<section>`s, including `site header`, `site footer`, and document structure. The article itself has a header, as do each of the sections.

Reviewer: ```suggestion
The `<section>` element is used to encompass generic standalone sections of a document when there is no more specific semantic element to use. Sections should have a heading, with very few exceptions.  Going back to the Jacinda Ardern example, on the home page of the newspaper, the banner would include the name of the newspaper, followed by a single `<main>`, divided into several `<section>`s, each with a header, such as "World news" and "Politics"; and in each section you'll find a series of `<article>`s. Within each `<article>`, you might find one or more `<section>` elements as well. If you look at this page, the entire "headers and sections" part is the `<article>`. This `<article>` is then divided into several `<section>`s, including `site header`, `site footer`, and document structure. The article itself has a header, as do each of the sections.
```

Some screen reader users do access headings to understand a page's content. Originally, headings were supposed to outline a document, just as MS Word or Google Docs can produce an outline based on headings, but browsers never implemented this structure. While browsers do display nested headings at increasingly smaller font sizes as shown in the following example, they don't actually support outlining.

{% Codepen {
  user: 'web-dot-dev',
  id: 'oNdePZj',
  height: 500,
  theme: 'dark',
  tab: 'html,result'
} %}

You now have enough knowledge to outline MachineLearningWorkshop.com:

## Outlining the `<body>` of MLW.com

This is the outline for the visible content of the machine learning workshop site:

{% Codepen {
  user: 'web-dot-dev',
  id: 'gOzxdxR',
  height: 700,
  theme: 'dark',
  tab: 'html'
} %}

{% Aside %}  
This code snippet only includes the content of the `<body>`. The [`<!doctype>`, `<html>`, `<body>`, and meta-information](/learn/html/metadata/) were covered earlier.  
{% endAside%}

As no piece of content is a standalone, complete piece of content, `<section>` is more appropriate than `<article>`; while each has a heading, no section is worthy of a `<footer>`.

> It should go without saying by this point, but don't use headings to make text bold or large; use CSS instead. If you want to emphasize text, there are semantic elements to do that too. We'll cover that in the next section. In fact, we'll fill in most of the page's content as we discuss text basics.

Reviewer: ```suggestion
It should go without saying by this point, but don't use headings to make text bold or large; use CSS instead. If you want to emphasize text, there are semantic elements to do that too. We'll cover that and fill in most of the page's content as we discuss text basics; after taking a deeper dive into attributes.
```

attributes comes next, before text basics
